hbase-server/src/main/java/org/apache/hadoop/hbase/master/cleaner/CleanerChore.java

```diff
@@ -322,140 +330,154 @@ public boolean getEnabled() {
     T act() throws IOException;
   }
 
-  /**
-   * Attemps to clean up a directory, its subdirectories, and files. Return value is true if
-   * everything was deleted. false on partial / total failures.
-   */
-  private final class CleanerTask extends RecursiveTask<Boolean> {
-    private static final long serialVersionUID = -1584635903138015418L;
-    private final Path dir;
-    private final boolean root;
+  private interface Callback {
```

Comment:
Can `Callback` be generic?

Reply:
For me, the callabck & AsyncResult can only handle the boolean type, so  it's OK if we don't make it generic.  Of course, if want to make the same behavior as Action<T> , then maybe it can be generic.